myhpom/templates/myhpom/accounts/signup.html

```diff
@@ -20,68 +20,95 @@ <h5 class="pseudo-modal__title--centered">Already have an account? <a href="{% u
+                                        {% if form.errors.state %}{{ form.errors.state }}{% endif %}
+                                    </div>
                                 </div>
-                                <div class="col-md-6 form-group">
-                                    <label for="id_password">Enter a password</label>
-                                    <input type="password" class="form-control" name="password" id="id_password" />
+                                <div class="col-md-6 form-group">  
+                                    <label class="{% if form.errors.password %}form-group__label--invalid{% endif %}" for="id_password">{{form.fields.password.label|upper}}</label>
+                                    <input type="password" class="form-control{% if form.errors.password %} is-invalid{% endif %}" name="password" id="id_password" value="{% if form.data.password %}{{form.data.password}}{% endif %}" />
+                                    <div class="invalid-feedback">
+                                        {% if form.errors.password %}{{ form.errors.password }}{% endif %}
+                                    </div>
                                 </div>
 
                                 <div class="col-md-6 form-group">
-                                    <label for="id_password_confirm">Confirm password</label>
-                                    <input type="password" class="form-control" name="password_confirm" id="id_password_confirm" />
+                                    <label class="{% if form.errors.password_confirm %}form-group__label--invalid{% endif %}" for="id_password_confirm">{{form.fields.password_confirm.label|upper}}</label>
+                                    <input type="password" class="form-control{% if form.errors.password_confirm %} is-invalid{% endif %}" name="password_confirm" id="id_password_confirm" value="{% if form.data.password_confirm %}{{form.data.password_confirm}}{% endif %}" />
+                                    <div class="invalid-feedback">
+                                        {% if form.errors.password_confirm %}{{ form.errors.password_confirm }}{% endif %}
+                                    </div>
                                 </div>
                             </div>
                         </div>
                         <div class="col-lg-4">
                             <div class="row">
-                                <div class="col-12 form-group form-group-tos">
+                                <div class="col-12 form-group form-group-tos {% if form.errors.accept_tos %}form-group-tos__div--invalid{% endif %}">
                                     <div class="form-group-tos__content">
-                                        <h5>Terms and conditions</h5>
+                                        <h5 class="{% if form.errors.accept_tos %}form-group-tos__h5--invalid{% endif %}">Terms and conditions</h5>
```

Comment:
When `form.errors.accept_tos` is true, there are now two boxes: one from `form-group-tos__content` (gray) and one from this new class (red). I'd suggest moving this if-condition up one level in the DOM and renaming it `form-group-tos__content--invalid`, and switch between these two box classes.